sklearn/impute/_base.py

```diff
@@ -440,21 +450,28 @@ def transform(self, X):
                                  "== 0 and input is sparse. Provide a dense "
                                  "array instead.")
             else:
-                mask = _get_mask(X.data, self.missing_values)
+                # if invalid statistics are eliminated recompute mask
+                # else use the mask compued before eliminating invalid mask
+                if not isinstance(valid_statistics_indexes, slice):
```

Comment:
and here `if valid_statistics_indexes is not None`. It would be better than using some type checking which is not intuitive why this type.